pkg/scheduler/drf_preemption_policy.go

```diff
@@ -114,14 +116,14 @@ type singleNodePreemptResult struct {
 // Do surgical preemption on node, if able to preempt, returns
 func trySurgicalPreemptionOnNode(preemptionPartitionCtx *preemptionPartitionContext, preemptorQueue *preemptionQueueContext, node *SchedulingNode, candidate *SchedulingAllocationAsk,
     headroomShortages map[string]*resources.Resource) *singleNodePreemptResult {
-    // To preempt resource = (allocating + candidate.asked) - (preempting + available)
-    resourceToPreempt := resources.Add(node.AllocatingResource, candidate.AllocatedResource)
-    resourceToPreempt.SubFrom(node.PreemptingResource)
-    resourceToPreempt.SubFrom(node.CachedAvailableResource)
-    resourceToPreempt = resources.ComponentWiseMax(resourceToPreempt, resources.Zero)
+    // the number of resources to preempt for this request is the available - requested
+    // ignoring anything below 0 as we have more available than requested
+    // don't count at what is already marked for preemption (that is still used)
+    // the scheduling node's available resource takes into account what is being allocated
+    resourceToPreempt := resources.SubEliminateNegative(node.getAvailableResource(), candidate.AllocatedResource)
```

Comment:
resourceToPreempt should be (requested - available), right?

Reply:
Yes you're right fixed that